pypicloud/views/simple.py

```diff
@@ -31,13 +32,23 @@ def upload(request, content, name=None, version=None):
             return request.forbid()
         try:
             return request.db.upload(content.filename, content.file, name=name,
-                                     version=version)
+                                     version=version, summary=summary)
         except ValueError as e:
             return HTTPBadRequest(*e.args)
     else:
         return HTTPBadRequest("Unknown action '%s'" % action)
 
 
+@xmlrpc_method(endpoint='pypi')
+def search(request, criteria, query_type):
```

Comment:
I also wanted to throw in a note about this - I have separately been working on configuring the search to also search the fallback url (`pypi.fallback` setting) and return those results so it is not limited to just packages uploaded to the local pypi server. I think I have it working fairly well, but I'll do a separate PR for it.